logbook/logbook.go

```diff
@@ -861,6 +861,15 @@ func (book *Book) RemoveLog(ctx context.Context, sender identity.Author, ref dsr
 	return book.save(ctx)
 }
 
+// RemoveAnyoneLog removes a log owned by anyone.
+func (book *Book) RemoveAnyoneLog(ctx context.Context, ref dsref.Ref) error {
+	if book == nil {
+		return ErrNoLogbook
+	}
+	book.store.RemoveLog(ctx, dsRefToLogPath(ref)...)
+	return book.save(ctx)
+}
```

Comment:
Ah I see, a bunch of this confusion is my fault. I wrote the current `logbook.RemoveLog` at the same time as `logsync`, and was attempting to prevent situations where foreign users shouldn't be allowed remove logs they don't have write access to. The _only_ place this is needed is when operating as a remote that supports delete requests

Given that you've just had to write a new `RemoveLog` that doesn't have this requirement, clearly we need a method that allows arbitrary log removal. To me this means `RemoveAnyoneLog` is the "real" `RemoveLog`, and we should move the author-only check to either a method on logbook that's called on `logbook/logsync/logsync.go#L273`, or externalized to that function.

Either way, in the name of a better API, we should rename `RemoveAnyoneLog` to `RemoveLog`. That's what it's doing.